From 0b388ae310192f7c9e8e4dfddb3c53c3873a98ea Mon Sep 17 00:00:00 2001
From: Liangtao Gao <gliangtao@gmail.com>
Date: Wed, 16 Oct 2024 09:55:14 -0700
Subject: [PATCH 01/16] Settings crash on devices without fingerprint feature

Bug: https://issuetracker.google.com/issues/374010061

Analysis:
fingerprint feature is optional, so Settings code should check
if it's supported before trying to get FingerprintManager.

Test: manual

Change-Id: Id58986ce5c12914eb226f6f8a51c2f5a36c26595
Signed-off-by: Liangtao Gao <gliangtao@gmail.com>
---
 .../settings/biometrics/fingerprint2/BiometricsEnvironment.kt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/biometrics/fingerprint2/BiometricsEnvironment.kt b/src/com/android/settings/biometrics/fingerprint2/BiometricsEnvironment.kt
index 215692a..133ce3f 100644
--- a/src/com/android/settings/biometrics/fingerprint2/BiometricsEnvironment.kt
+++ b/src/com/android/settings/biometrics/fingerprint2/BiometricsEnvironment.kt
@@ -24,6 +24,7 @@ import androidx.lifecycle.ViewModelStore
 import androidx.lifecycle.ViewModelStoreOwner
 import com.android.internal.widget.LockPatternUtils
 import com.android.settings.SettingsApplication
+import com.android.settings.Utils
 import com.android.settings.biometrics.GatekeeperPasswordProvider
 import com.android.settings.biometrics.fingerprint2.data.repository.DebuggingRepository
 import com.android.settings.biometrics.fingerprint2.data.repository.DebuggingRepositoryImpl
@@ -74,8 +75,7 @@ class BiometricsEnvironment(context: SettingsApplication) : ViewModelStoreOwner
   private val backgroundDispatcher = executorService.asCoroutineDispatcher()
   private val applicationScope = MainScope()
   private val gateKeeperPasswordProvider = GatekeeperPasswordProvider(LockPatternUtils(context))
-  private val fingerprintManager =
-    context.getSystemService(FragmentActivity.FINGERPRINT_SERVICE) as FingerprintManager?
+  private val fingerprintManager = Utils.getFingerprintManagerOrNull(context)
 
   private val fingerprintSensorRepository: FingerprintSensorRepository =
     FingerprintSensorRepositoryImpl(fingerprintManager, backgroundDispatcher, applicationScope)
-- 
2.47.1

